unblob/handlers/archive/qnap/qnap_nas.py

```diff
@@ -58,6 +58,7 @@ def is_valid_header(header: Instance) -> bool:
 def _hyperscan_match(
     context: QTSSearchContext, pattern_id: int, offset: int, end: int
 ) -> Scan:
+    del pattern_id, end
```

Comment:
A couple of these `del`s  were introduced due to [flake8-unused-arguments](https://beta.ruff.rs/docs/rules/#flake8-unused-arguments-arg) rule.

I have mixed feelings abut this. One thing is that I like this rule, it sounds a good thing to have. On the other hand,  using explicit `del` statement feels "weird". I know this is very opinionated but I can not phrase it better.

Is there anything else we can do here?  I would say just don't provide those parameters, but they must be there for a reason (hyperscan need that?)

If the code can not be changed that way, what else can we do? I found out starting the parameter name with underscore (`_pattern_id`, `_end` in this case) would also solve this problem (not trigger ruff). 

For me, it "looks" better - but as I was saying, this is very subjective so I can be convinced leaving this.

Reply:
Renaming parameters breaks the interface though, as it can no longer called with keyword arguments.

Let's continue at https://github.com/onekey-sec/unblob/pull/540#discussion_r1152900184